defopt.py

```diff
@@ -158,7 +158,10 @@ class _NoTypeFormatter(_Formatter):
 
 
 def _populate_parser(func, parser, parsers, short):
-    sig = _inspect_signature(func)
+    full_sig = _inspect_signature(func)
+    sig = full_sig.replace(
+        parameters=[param for param in full_sig.parameters.values()
+                    if not param.name.startswith('_')])
```

Comment:
Can I get you to use `list(...)` here for the same reason you have `dict` and `set` below?